app/models/task.rb

```diff
@@ -22,64 +22,12 @@ class Task < ApplicationRecord
   accepts_nested_attributes_for :tests, allow_destroy: true
   accepts_nested_attributes_for :model_solutions, allow_destroy: true
 
-  scope :text_like, lambda { |text|
-    if text.present?
-      where('title ILIKE ? OR description ILIKE ?', "%#{text}%", "%#{text}%")
-    else
-      where(nil)
-    end
-  }
-  scope :mine, lambda { |user|
-    if user.nil?
-      where(nil)
-    else
-      where(user_id: user.id)
-    end
-  }
+  scope :visibility, ->(visibility, user = nil) { {owner: where(user: user), public: all}.with_indifferent_access[visibility] }
```

Comment:
You are probably referring to this line in your comment, aren't you? I was really thinking about that and it also took me quite some time to understand it (there is some cleverness involved!). 

What about the "private" exercises? I remember that we discussed these are misleading (because of the missing differentiation between "private" and "own" tasks). Is this why you are not filtering for these here? And why can we use `all` for public where "private" / "own" exercises would be included as well? I fear, I miss something here (or is this missing and part of another MR?) Sorry for asking these (stupid) questions...

Reply:
Currently I'm only differentiating between "own" and "all" tasks. There is option yet to make a task private or something similar. Excluding own tasks during the public search makes perfect sense - I will fix that.